### ISSUE TITLE
Changing a role from being "online" to "offline" does not also change status of existing users having this role

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#8668]
**Describe the bug**
Users can have a specific role flag `mm-online`, which determines whether they are online or offline users. This role is added or removed when the user is created or edited, depending on the list of roles that is provided for them. 
If a user is assigned to an role that is `online`, they will also be assigned this `mm-online` role. 
If the user's primary role is later changed, from being online to offline, any user that has this role will still be treated as `online`, because of the `mm-online` flag. 

**To Reproduce**
1. Create a new online role. 
2. Create a user with this online role. 
3. Edit the role and make it offline. 
4. Log in with the user and observe they are treated as an online user: they have access to all documents and do not have the app available when they are offline. 

**Expected behavior**
It should be clear that simply editing the role is not sufficient to changing the behavior of users that are assigned to this role. 
An alternative would be to edit the users every time a role is changed. 

**Environment**
- Instance: self hosted partner
- Version: all

**Additional context**
This came in through the forum: https://forum.communityhealthtoolkit.org/t/scope-report-viewing-to-specific-top-level-place/3092/9

**Comment by dianabarsan** [comment]
I don't believe this will be a straight forward change so all users are edited when a role is changed kind of situation. 
Changing a user from being online to being offline, and vice-versa, while the user is logged in will produce all sorts of unexpected results, so propagating the role change to users can be seriously risky. 

Not providing any indication that the users won't be affected by the role edit needs to change.

**Comment by garethbowen** [comment]
Yeah this sounds risky. The quick fix would be to block anyone switching, essentially forcing creating a new user. I'm not sure how common it would be to actually want to change an existing user...

**Comment by dianabarsan** [comment]
Oh, interesting idea. So switching a role online status would disable all user logins that have this role? 
Because roles are likely changed through pushing config, instead of the admin UI (I think we don't even offer this switch as a UI option), we would need to have some sort of warning in cht-conf.

**Comment by Abhishek9639** [comment]
Hi @dianabarsan,
I would like to work on this issue. I’ve gone through the issue description and would be interested in investigating and fixing it. Could you please assign this issue to me?
Thanks!

**Comment by jkuester** [comment]
I will defer to @dianabarsan for the final call here, but I do not think we have a clear design path for how to address this yet. 😓   

I think we definitely should not support editing the `online` status of an existing role.  This would be easy to validate (e.g. add a new settings validation framework to the [settings service](https://github.com/medic/cht-core/blob/master/api/src/services/settings.js)).  However, that does not really cover the _core_ of the issue right?  The roles assigned to a user _can_ change (and we want to support this).  So, even if the actual config for the role does not change, a user created with an `online` role might be edited later to only have offline roles.  I have no idea how to handle that case (except for maybe to just reject the edit???)

**Comment by Abhishek9639** [comment]
Hey @jkuester or @dianabarsan,
I understand the concern, especially around users being edited later to only have offline roles even if they were initially created with an online role. I agree that simply validating the role config would not fully solve the core issue. I’ll wait for further direction once there’s a clearer design approach decided.
Thanks!

**Comment by dianabarsan** [comment]
Given how differently the webapp handles online vs offline workflows, I think we will never actually support the switch for a user. So, I don't think it's about blocking the edit here necessarily, but actually making the fixture obvious and very intentional. 

As in: right now the status of the user being online or offline is decided on the fly based on current configuration, and I think it maybe should not? Maybe this should be something that is mandated by the user document and is never allowed to be changed. 

Right now, such distinction exists but barely through the `mm-online` role. 

Sorry @Abhishek9639, these are just my random thoughts and not a specific technical direction or design.
